Dependencies update, code fixes, travis configuration fix

christianascone opened the pull request:
What I did:

- Updated Pod Opera spec (`Opera` to `OperaSwift`) with 2.0 version
- Changed every Opera reference (Opera to OperaSwift)
- Added missing URLRequestSetup protocol
- Fixed OTHER_SWIFT_FLAGS warning for Release config (Added `$(inherited)` flag)
- Updated RxSwift examples with new name convetion (for example `rx_object()` to `rx().object()`). I suggest you to take a look at this.
- Fixed travis configuration updating gems versions and Xcode version

christianascone committed:
Fix wrong Pod spec for Opera

Changed Opera to OperaSwift and updated the branch specification using the 2.0 release

christianascone committed:
Fix OTHER_SWIFT_FLAGS warning during pod installation

Pod install was thrown `target overrides the `OTHER_SWIFT_FLAGS` build setting` warning due to absence of `$(inherited)` flag in release for both the project targets.

christianascone committed:
Update pod’s Opera reference in README

According to Opera -> OperaSwift update in pod spec (https://github.com/xmartlabs/Opera/commit/52b52643db9a96ea8e462a25dad2f02638db5d09), it has been updated in this project’s Podfile.

christianascone committed:
Fix Opera module import to OperaSwift

christianascone committed:
Add URLRequestSetup protocol

christianascone committed:
Update rx user login to new name convention

christianascone committed:
Update Copyright symbol for some classes ( (c) to ©)

christianascone committed:
Update Xcode version in travis configuration file

christianascone committed:
Add bundle install for missing gems

christianascone committed:
Add cocoa pods installation

christianascone committed:
Update gem lock file

christianascone committed:
Merge branch 'feature/travis_config_fix' into develop

m-revetria commented on Project-iOS/XLProjectName/XLProjectName/Networking/Manager.swift:
```diff
@@ -18,18 +18,18 @@ class NetworkManager: RxManager {
-        return SessionController.sharedInstance.refreshToken().flatMap { _ in response }
+    override func response(_ requestConvertible: URLRequestConvertible) -> PrimitiveSequence<SingleTrait, OperaResult> {
+        let response = super.response(requestConvertible)
+        return response
```
Please keep the call to the `refreshToken` function

m-revetria reviewed (commented):
Hi @christianascone! Thanks for the contribution it looks really good. I left just a comment on. Please take a look and we will be glad to merge this PR.

christianascone committed:
Add call to `refreshToken` in Manager's response function

The call to refreshToken has been re-added and its result is converted into Single<OperaResult> so the function can be compliant to overridden superclass' function.

christianascone commented:
Thank you @m-revetria !
I just restored the call to the `refreshToken` function and checked Travis-CI status (build [passed](https://travis-ci.org/xmartlabs/Swift-Project-Template/builds/274804528?utm_source=github_status&utm_medium=notification)).

m-revetria reviewed (approved):
@christianascone thanks for fixing the comment!
